reader.go

```diff
@@ -436,7 +439,9 @@ func (r *Reader) syncGroup(conn *Conn, memberAssignments GroupMemberAssignments)
 
 	if len(assignments.Topics) == 0 {
 		generation, memberID := r.membership()
-		return nil, fmt.Errorf("received empty assignments for group, %v as member %s for generation %d", r.config.GroupID, memberID, generation)
+		r.withLogger(func(l *log.Logger) {
```

Comment:
If I'm reading this correctly, this is the only change required to address #200.

Could we make this a ReaderOption `AllowEmptyAssignments`? That would allow the option of preserving existing behavior for the non-existent topic case while allowing us to move forward with allowing more consumers than topics.

Reply:
There are a few downsides to consider there:
- It would be public, and therefore hard to get rid of.
- I'd argue it should default to true, to avoid a risky, subtle, and possibly common situation. 
- Too much config is a bad thing, IMO.

Reply:
I prefer not to have a configuration option.  Currently, the code is incorrect because there are valid use cases and configurations that _should_ work but don't.  The downside of allowing empty assignments is that perhaps someone accidentally spins up too many consumers and some number of them sit idle.  Preventing things like that feels like it's outside the scope of this library.

Reply:
Those are good arguments for not making it an option.  Consider my suggestion withdrawn.